src/ui/components/Avatar.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+
+import { connect } from "react-redux";
+import { selectors } from "ui/reducers";
+import { actions } from "ui/actions";
+import { getAvatarColor } from "ui/utils/user";
+
+const Avatar = props => {
+  let { player, isFirstPlayer, updateUser, isLoggedIn, setLoggedIn, setLoggedOut } = props;
+  let auth = useAuth0();
+
+  if (auth.isAuthenticated && isFirstPlayer) {
+    if (!player.name) {
+      // Check if the user has just logged out. If so, update and add the associated
+      // picture and name from the user metadata.
+      updateUser(auth.user);
+    }
```

Comment:
Hmm, i'm wondering if this should be in a `useEffect`. Not too sure, i don't know hooks :)


```js
useEffect(() => updateUser(auth.user), [auth, isFirstPlayer])
```